File: codegen/results/v002/m3-max-48gb/lmstudio/qwen3.8-27b-gguf-q4km_nothink_hacked/20260830_221322/typescript/005_unit_test_writer/solution.ts
function calculateDiscount(price: number, quantity: number): number {
    if (quantity < 10) return price * quantity;
    else if (quantity < 50) return price * quantity * 0.9;
    else return price * quantity * 0.75;
}

interface TestCase {
    description: string;
    price: number;
    quantity: number;
    expected: number;
}

const testCases: TestCase[] = [
    {
        description: "quantity 9 (below 10, no discount)",
        price: 10,
        quantity: 9,
        expected: 10 * 9,
    },
    {
        description: "quantity 10 (boundary, 10% discount)",
        price: 10,
        quantity: 10,
        expected: 10 * 10 * 0.9,
    },
    {
        description: "quantity 49 (boundary, 10% discount)",
        price: 10,
        quantity: 49,
        expected: 10 * 49 * 0.9,
    },
    {
        description: "quantity 50 (boundary, 25% discount)",
        price: 10,
        quantity: 50,
        expected: 10 * 50 * 0.75,
    },
    {
        description: "quantity 100 (well above 50, 25% discount)",
        price: 20,
        quantity: 100,
        expected: 20 * 100 * 0.75,
    },
];

function runTests(): void {
    for (const tc of testCases) {
        const result = calculateDiscount(tc.price, tc.quantity);
        const tolerance = 1e-10;
        if (Math.abs(result - tc.expected) <= tolerance) {
            console.log(`PASS: ${tc.description}`);
        } else {
            console.log(`FAIL: ${tc.description} (expected: ${tc.expected}, got: ${result})`);
        }
    }
}

runTests();

File: codegen/results/v002/m3-max-48gb/lmstudio/qwen3.8-27b-gguf-q4km_nothink_hacked/20260830_221322/python/005_unit_test_writer/solution.py
def calculate_discount(price: float, quantity: int) -> float:
    """
    Calculate total price after bulk discount.
    quantity < 10: no discount
    quantity 10-49: 10% off
    quantity >= 50: 25% off
    Returns: price * quantity * discount_multiplier
    """
    if quantity < 10:
        return price * quantity
    elif quantity < 50:
        return price * quantity * 0.9
    else:
        return price * quantity * 0.75


def test_calculate_discount():
    test_cases = [
        # (price, quantity, description)
        (10.0, 9, "Boundary: quantity 9 (no discount)"),
        (10.0, 10, "Boundary: quantity 10 (10% discount)"),
        (10.0, 49, "Boundary: quantity 49 (10% discount)"),
        (10.0, 50, "Boundary: quantity 50 (25% discount)"),
        (5.0, 5, "Below discount threshold"),
        (5.0, 25, "Mid-range discount"),
        (5.0, 100, "High quantity discount"),
        (0.0, 10, "Zero price"),
        (100.0, 1, "Single item no discount"),
    ]
    
    for price, quantity, desc in test_cases:
        result = calculate_discount(price, quantity)
        
        # Calculate expected value
        if quantity < 10:
            expected = price * quantity
        elif quantity < 50:
            expected = price * quantity * 0.9
        else:
            expected = price * quantity * 0.75
        
        # Use a small tolerance for floating point comparison
        if abs(result - expected) < 1e-9:
            print(f"PASS: {desc}")
        else:
            print(f"FAIL: {desc} (expected: {expected}, got: {result}")


if __name__ == "__main__":
    test_calculate_discount()